Foundation/FileManager.swift

```diff
@@ -693,11 +693,10 @@ open class FileManager : NSObject {
     /* Process working directory management. Despite the fact that these are instance methods on FileManager, these methods report and change (respectively) the working directory for the entire process. Developers are cautioned that doing so is fraught with peril.
      */
     open var currentDirectoryPath: String {
-        let length = Int(PATH_MAX) + 1
-        var buf = [Int8](repeating: 0, count: length)
-        getcwd(&buf, length)
-        let result = self.string(withFileSystemRepresentation: buf, length: Int(strlen(buf)))
-        return result
+        let len = Int(PATH_MAX) + 1
+        var buf = [Int8](repeating: 0, count: len)
+        getcwd(&buf, len)
+        return string(withFileSystemRepresentation: buf, length: len)
     }
```

Comment:
`string(withFileSystemRepresentation:length:)` takes the length of the `NUL`-terminated string, not the whole buffer so `strlen(buf)` is correct here - although `destinationOfSymbolicLink(atPath:)` seems to be written the wrong way round so needs fixing.